xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiHibernateStore.java

```diff
@@ -2063,9 +2063,16 @@ private void releaseAllLocksForCurrentUser(final XWikiContext ctx)
             Session session = getSession(context);
 
             // the select clause is compulsory to reach the fullName i.e. the page pointed
+            // here we check both the backlink serialized with AND without the wiki part
+            // the reason is in case of a farm of wiki, if the link points to a page from another wiki
+            // its reference is saved with the wiki part, so using only the local serialization won't be enough in that
+            // case. This should be changed once the refactoring to support backlinks properly has been done.
+            // See: XWIKI-16192
             Query query = session
-                .createQuery("select backlink.fullName from XWikiLink as backlink where backlink.id.link = :backlink");
+                .createQuery("select backlink.fullName from XWikiLink as backlink where "
+                    + "backlink.id.link in (:backlink, :backlinkwithwiki)");
```

Comment:
This logic is wrong. You will end up with links having nothing to do with the document you are interrested in (documents with the same fullname than the document in a different wiki).

Instead you should do a different search depending if you are requesting the document's wiki or not.